### DISCUSSION TITLE
Differentiation of access rights

### DISCUSSION THREAD

**Discussion by Pitcentr** [directus/directus#13100] [Archive]
How can I prevent the search method from viewing orders of other users ?
How can I check whether an order belongs to a user and only then give an API response ?

**Comment by rijkvanzanten** [comment]
That's where permissions come in. See https://docs.directus.io/configuration/users-roles-permissions/#configuring-permissions for more information 👍🏻

**Reply by Pitcentr** [reply]
Thank you, I know that
I have a collection of "orders"
/items/orders?filter[photocenter_id]=14
and the backend gives requests of the form, but the visitor can intercept the request and change it to an example
/items/orders?filter[photocenter_id]=15
and
/items/orders?filter[photocenter_id]=16
etc
how to implement protection against this ?

**Reply by Pachat** [reply]
For my part, I always use `UUID` for `id` instead of `integer`

**Reply by azrikahar** [reply]
@Pitcentr By configuring the permissions properly as mentioned (such as `photocenter_id` equals their ID, this is just a simple example, do refer the docs for more info), the visitor will still not be able to view those orders despite their enumeration attempts.
